hack/generated/controllers/generic_controller.go

```diff
@@ -355,8 +358,13 @@ func (gr *GenericReconciler) MonitorDelete(
 	}
 
 	// already deleting, just check to see if it still exists and if it's gone, remove finalizer
-	found, err := gr.ARMClient.HeadResource(ctx, resource.GetId(), resource.Spec().GetApiVersion())
+	found, err, retryAfter := gr.ARMClient.HeadResource(ctx, resource.GetId(), resource.Spec().GetApiVersion())
 	if err != nil {
+		if retryAfter != 0 {
+			data.log.Info("Error HEADing resource, will retry", "delay", retryAfter/time.Second)
```

Comment:
need to log error as we are hiding it?

Reply:
Error performing HEAD on resource, will retry? (HEADing reads sorta weird?)